From e84fab8d10cddfcb9d423b0192633da200bf1642 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 8 Feb 2023 12:22:39 +0100
Subject: [PATCH 0345/1695] drivers: stm32_exti: let exported function use
 private parameter

Now that all the users pass the proper private parameter, let all
the exports functions use the private parameter rather than the
global variable.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e5ba63e8a3fa3ecc91cee10500901157e44a8ef1 drivers: stm32_exti: let exported function use private parameter]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ia32d4944e9d3c9f968522db1ce00635f207eb314
---
 core/drivers/stm32_exti.c | 85 ++++++++++++++++++---------------------
 1 file changed, 40 insertions(+), 45 deletions(-)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index 4b8a0d4a5..d9004d870 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -55,8 +55,8 @@ static uint32_t stm32_exti_get_bank(uint32_t exti_line)
 	return bank;
 }
 
-void stm32_exti_set_type(struct stm32_exti_pdata *exti __unused,
-			 uint32_t exti_line, uint32_t type)
+void stm32_exti_set_type(struct stm32_exti_pdata *exti, uint32_t exti_line,
+			 uint32_t type)
 {
 	uint32_t mask = BIT(exti_line % 32);
 	uint32_t r_trig = 0;
@@ -83,15 +83,15 @@ void stm32_exti_set_type(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	io_mask32(stm32_exti.base + _EXTI_RTSR(i), r_trig, mask);
-	io_mask32(stm32_exti.base + _EXTI_FTSR(i), f_trig, mask);
+	io_mask32(exti->base + _EXTI_RTSR(i), r_trig, mask);
+	io_mask32(exti->base + _EXTI_FTSR(i), f_trig, mask);
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_mask(struct stm32_exti_pdata *exti __unused, uint32_t exti_line)
+void stm32_exti_mask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t val = 0;
 	uint32_t mask = BIT(exti_line % 32);
@@ -100,18 +100,17 @@ void stm32_exti_mask(struct stm32_exti_pdata *exti __unused, uint32_t exti_line)
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	val = io_read32(stm32_exti.base + _EXTI_C1IMR(i));
+	val = io_read32(exti->base + _EXTI_C1IMR(i));
 	val &= ~mask;
-	io_write32(stm32_exti.base + _EXTI_C1IMR(i), val);
-	stm32_exti.mask_cache[i] = val;
+	io_write32(exti->base + _EXTI_C1IMR(i), val);
+	exti->mask_cache[i] = val;
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_unmask(struct stm32_exti_pdata *exti __unused,
-		       uint32_t exti_line)
+void stm32_exti_unmask(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t val = 0;
 	uint32_t mask = BIT(exti_line % 32);
@@ -120,18 +119,17 @@ void stm32_exti_unmask(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	val = io_read32(stm32_exti.base + _EXTI_C1IMR(i));
+	val = io_read32(exti->base + _EXTI_C1IMR(i));
 	val |= mask;
-	io_write32(stm32_exti.base + _EXTI_C1IMR(i), val);
-	stm32_exti.mask_cache[i] = val;
+	io_write32(exti->base + _EXTI_C1IMR(i), val);
+	exti->mask_cache[i] = val;
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_enable_wake(struct stm32_exti_pdata *exti __unused,
-			    uint32_t exti_line)
+void stm32_exti_enable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % 32);
 	uint32_t exceptions = 0;
@@ -139,15 +137,14 @@ void stm32_exti_enable_wake(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	stm32_exti.wake_active[i] |= mask;
+	exti->wake_active[i] |= mask;
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_disable_wake(struct stm32_exti_pdata *exti __unused,
-			     uint32_t exti_line)
+void stm32_exti_disable_wake(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % 32);
 	uint32_t exceptions = 0;
@@ -155,15 +152,14 @@ void stm32_exti_disable_wake(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	stm32_exti.wake_active[i] &= ~mask;
+	exti->wake_active[i] &= ~mask;
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_clear(struct stm32_exti_pdata *exti __unused,
-		      uint32_t exti_line)
+void stm32_exti_clear(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % 32);
 	uint32_t exceptions = 0;
@@ -171,16 +167,15 @@ void stm32_exti_clear(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	io_mask32(stm32_exti.base + _EXTI_RPR(i), mask, mask);
-	io_mask32(stm32_exti.base + _EXTI_FPR(i), mask, mask);
+	io_mask32(exti->base + _EXTI_RPR(i), mask, mask);
+	io_mask32(exti->base + _EXTI_FPR(i), mask, mask);
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_set_tz(struct stm32_exti_pdata *exti __unused,
-		       uint32_t exti_line)
+void stm32_exti_set_tz(struct stm32_exti_pdata *exti, uint32_t exti_line)
 {
 	uint32_t mask = BIT(exti_line % 32);
 	uint32_t exceptions = 0;
@@ -188,15 +183,15 @@ void stm32_exti_set_tz(struct stm32_exti_pdata *exti __unused,
 
 	i = stm32_exti_get_bank(exti_line);
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	io_mask32(stm32_exti.base + _EXTI_SECCFGR(i), mask, mask);
+	io_mask32(exti->base + _EXTI_SECCFGR(i), mask, mask);
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
-void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti __unused,
-				  uint8_t bank, uint8_t pin)
+void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti, uint8_t bank,
+				  uint8_t pin)
 {
 	uint32_t reg = _EXTI_CR(pin / 4);
 	uint32_t shift = (pin % 4) * 8;
@@ -204,11 +199,11 @@ void stm32_exti_set_gpio_port_sel(struct stm32_exti_pdata *exti __unused,
 	uint32_t mask = 0xff << shift;
 	uint32_t exceptions = 0;
 
-	exceptions = cpu_spin_lock_xsave(&stm32_exti.lock);
+	exceptions = cpu_spin_lock_xsave(&exti->lock);
 
-	io_mask32(stm32_exti.base + reg, val, mask);
+	io_mask32(exti->base + reg, val, mask);
 
-	cpu_spin_unlock_xrestore(&stm32_exti.lock, exceptions);
+	cpu_spin_unlock_xrestore(&exti->lock, exceptions);
 }
 
 static void stm32_exti_pm_suspend(void)
-- 
2.47.2

